File: tests/output/jpa/src/main/javagen/restaurant/jpa_server/dtos/restaurant/PlatItemReadonly.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_server.dtos.restaurant;

import java.math.BigDecimal;

import jakarta.annotation.Generated;

import restaurant.jpa_server.enums.restaurant.CategoriePlatCode;

/**
 * Détail d'un plat en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface PlatItemReadonly {

	/**
	 * Identifiant du plat.
	 */
	Integer getId();

	/**
	 * Nom du plat.
	 */
	String getNom();

	/**
	 * Catégorie du plat.
	 */
	CategoriePlatCode getCategoriePlatCode();

	/**
	 * Prix du plat.
	 */
	BigDecimal getPrix();

	/**
	 * Indique si le plat est disponible.
	 */
	Boolean getDisponible();

	/**
	 * Prix du plat.
	 * @param prix value to set.
	 */
	void setPrix(BigDecimal prix);

	/**
	 * Indique si le plat est disponible.
	 * @param disponible value to set.
	 */
	void setDisponible(Boolean disponible);
}

File: tests/output/vanilla-js/model/restaurant/plat-item-readonly.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_BOOLEEN, DO_CODE, DO_LIBELLE, DO_PRIX, DO_SEQ_ID} from "../../domains";

import {CategoriePlatCode} from "./enums";

export interface PlatItemReadonly {
    id: number;
    nom: string;
    categoriePlatCode: CategoriePlatCode;
    prix: number;
    disponible: boolean;
}

export const PlatItemReadonlyEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_SEQ_ID,
        isRequired: true,
        label: "Id"
    },
    nom: {
        type: "field",
        name: "nom",
        domain: DO_LIBELLE,
        isRequired: true,
        label: "Nom"
    },
    categoriePlatCode: {
        type: "field",
        name: "categoriePlatCode",
        domain: DO_CODE,
        isRequired: true,
        label: "CategoriePlatCode"
    },
    prix: {
        type: "field",
        name: "prix",
        domain: DO_PRIX,
        isRequired: true,
        label: "Prix"
    },
    disponible: {
        type: "field",
        name: "disponible",
        domain: DO_BOOLEEN,
        defaultValue: true,
        isRequired: true,
        label: "Disponible"
    }
} as const;
